int8 window_size
---
bool success
string message